plugins/net_plugin/net_plugin.cpp

```diff
@@ -4007,7 +4018,8 @@ namespace eosio {
       });
    }
 
-   void net_plugin_impl::on_accepted_block() {
+   void net_plugin_impl::on_accepted_block( const signed_block_ptr& block, const block_id_type& id ) {
+      sync_master->send_handshakes_if_synced(fc::time_point::now() - block->timestamp);
```

Comment:
You can add an assert `now` is greater than `timestamp`.

Reply:
We can not assume that `now()` is greater than `block->timestamp`.

Reply:
Oh I see `fc::microseconds` checks negative input.